### ISSUE TITLE
Suggestion: Game-Rule for Setting Damage Values

### ISSUE THREAD

**Issue by Seimix** [Robosturm/Commander_Wars#512]
**Overview**
Now, this would be a Super Ambitious addition to the game, so I would not be upset if it gets rejected. But...
Basically, the request is to add an Advanced Option in the Game Rules menu that allows you to adjust individual unit damage values vs other units. (I.E, being able to set an infantry to deal 110% damage to a tank or whatever). And have it to where you can save those customized damage values in a similar way you would save Banlists for COs, Perks, and such. This would also allow you to have those other "Damage Formula" mods that come with the game just be pre-saved settings for the "Damage Value Editor". Example of what I mean: https://i.imgur.com/7DWvb7k.png 
(Got too lazy to fill out the entire image, but hopefully this gives you a good enough idea of what I mean) 
Note: Should probably also include Damage vs Terrain 

Whilst yes, you could already mod the game to change these values, this would drastically expand upon the customization this game offers in a much more accessible way and it would remove the need to make sure all players in a multiplayer match have the same mod that changes some Damage Values. 

Now, this does come at a cost though and that's being just how ambitious such an addition would be. Would not surprise me if it turns out to be too much work to do to incorporate. Other problems would also include: 

A. The Menu Getting too Cluttered if there are too many Units (Solutions to this would be to highlight the row and column of the damage value you're editing or giving the Rows and Columns alternating Colors)
B. The fact that damage vs units is based on the weapon a unit uses rather than the unit itself. (The only solution I have for this is using the most prominent unit of a weapon as the icon for it, and when hovering over that unit icon, it states all the other units that use the same weapon like this one.)
C. The fact on online matches, the other players would have NO IDEA what unit damage values have been changed and what hasn't. (Solution to this is to allow other players in an online match to be able to check the game's rules somehow and having the changed values be highlighted.)
D. Would probably overcomplicate the code and cause problems for modders.

Another somewhat problem is the fact that if this is possible to include, then why stop with Damage Values? Why not also include an in-game way to change the normal prices of all units or adjust the average luck value for everyone? (Although, I'm not as keen on including these as I am with this)

**TL; DR**
Basically: An option in the "Games Rules" Menu under the "Advanced" category that would allow you to set custom unit damage values for that match. Would look kind of like this: https://i.imgur.com/7DWvb7k.png (Should also probably include damage vs terrain) 
- The Ability to save the customized Damage Values like you would a ban list. 
-- On top of this, incorporating all the damage value mods that come with the game as just saved Damage Value Settings. (Unless I am HORRIBLY misunderstanding what those other damage formula mods even do)
- The Column and Row being highlighted while editing a unit's damage values. 
- The Ability for other players in a server to check the damage values set for that game
-- And having the edited damage values being highlighted 
- The most prominent unit/user of a weapon showing all the other units who use the same weapon when you hover over their icon. * (Not sure how necessary this is exactly, but I would imagine the description also mentioning which ammo type for a unit uses that weapon)

**Comment by Robosturm** [comment]
This is a feature which I can gurantee to be never happen. :(
The way the game is designed and how damage for weapons/units is created and what the c++ code should know about it.
Don't interact at all.
Furthermore i don't want to think about what it needs coding wise just to make it possible to work with vanilla thinks and restoring the JS-State before and after a match and the fact that you can mod in stuff which works without any damage tables make this suggestion not feasible.
